app/controllers/homepage_controller.rb

```diff
@@ -105,7 +111,17 @@ def find_listings(params, listings_per_page, filter_params)
     if numeric_search_needed && filter_params[:listing_id].empty?
       Listing.none.paginate(:per_page => listings_per_page, :page => params[:page])
     else
-      Listing.find_with(filter_params, @current_user, @current_community, listings_per_page, params[:page])
+      listings = Listing.find_with(filter_params, @current_user, @current_community, listings_per_page, params[:page]).map { |e|
```

Comment:
Should this whole block be now abstracted behind some service? Is this going to be the next phase?